src/App.tsx

```diff
@@ -1,14 +1,61 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { Options } from './types/Options';
+import { getTodos } from './api';
 import { Loader } from './components/Loader';
 
+const getFilteredTodos = (todos: Todo[], filter: Filter):Todo[] => {
+  let filterTodos = [...todos];
+
+  filterTodos = filterTodos.filter(todo => {
+    switch (filter.option) {
+      case Options.Active:
+        return !todo.completed;
+      case Options.Completed:
+        return todo.completed;
+      case Options.All:
+      default:
+        return todo;
+    }
+  });
+
+  if (filter.query) {
+    filterTodos = filterTodos
+      .filter(({ title }) => title.toLowerCase()
+        .includes(filter.query.toLowerCase()));
```

Comment:
Consider creating a variable for the lowercase query to avoid calling function for each element of the array